tajo-core/src/main/java/org/apache/tajo/master/QueryInfo.java

```diff
@@ -119,14 +119,37 @@ public int getQueryMasterClientPort() {
     return queryMasterClientPort;
   }
 
-  public TajoProtos.QueryState getQueryState() {
+  public synchronized TajoProtos.QueryState getQueryState() {
```

Comment:
queryState is volatile, so it does not need a lock. But for waitState method, I need a lock to synchronize. I just choosen "this" not to add any other object.